### PR TITLE
Bugfix/missing patch method in log switch

### LINKED ISSUE #147: [API300] Missing #patch in Logical Switch
### Scenario/Intent
I'm trying to find the Logical Switch resource patch operation but I can't.

### Environment Details
 - **OneView SDK Version:** 3.1.0
 - **OneView Client API Version:** 300 C7000

### PR DESCRIPTION
**Pull request by ricardogpsf**
### Description
Patch method was added to Logical Switch with helper methods to add, remove and replace scopes

### Issues Resolved
Fixes #147 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by ricardogpsf** [comment]
I'm working on this problem.

**Comment by jsmartt** [comment]
What is the relation between this and #148? Seems to be a lot of duplicate and things that aren't related to the issue called out that this PR fixes.

**Comment by ricardogpsf** [comment]
@jsmartt Sorry, I did not comment about this.

This PR need the Scope Resource, that is in #148 .

**Comment by ricardogpsf** [comment]
When #148 is merged, the changes will be just of this PR.
I do not know how I can create PR with dependencies (that need other commit in other PR) and Github know about this dependencies.
Do you know how we do this, @jsmartt ?

**Comment by jsmartt** [comment]
No, I'm not aware of any way to do that. Just with comments

**Comment by ricardogpsf** [comment]
Thanks for merge @jsmartt.

File: lib/oneview-sdk/resource/api300/c7000/logical_switch.rb [hunk 1]
```diff
@@ -45,6 +49,15 @@ def self.get_internal_link_set(client, name)
           results = get_internal_link_sets(client)
           results.find { |internal_link_set| internal_link_set['name'] == name }
         end
+
+        # Updates scopes uris of a given logical switch resource
+        # @param [String] operation The type of operation: one of "add", "remove" or "replace".
+        # @param [String] path The path operation for use. The exact meaning depends on the type of operation.
+        # @param [String] value The value to add or replace for "add" and "replace" operations. Not used by "remove".
+        # @note This attribute is subject to incompatible changes in future release versions, including redefinition or removal.
+        def patch(operation, path, value = nil)
```
Comment (tmiotto) [inline lib/oneview-sdk/resource/api300/c7000/logical_switch.rb]: I don't know if this updated comment would be helpful. I really think the user will have to look at the REST API to know how and what to put in this operation. Maybe we could remove this method from here, and leave it with the standard patch comment. :thought_balloon:

Comment (ricardogpsf) [inline lib/oneview-sdk/resource/api300/c7000/logical_switch.rb]: done. =)

Comment (jsmartt) [inline lib/oneview-sdk/resource/api300/c7000/logical_switch.rb]: Agreed. If it's a complicated operation that needs explanation, throw in an example.

File: lib/oneview-sdk/resource.rb [hunk 2]
```diff
@@ -327,12 +327,11 @@ def ensure_uri
     # @param [Array<OneviewSDK::Resource>] resources The list of resources
     # @return [Array<String>] List of uris
     # @raise IncompleteResource if 'uri' is not set for each resource.
-    def get_and_ensure_uri_for(resources)
-      uris = resources.map do |resource|
+    def get_and_ensure_uri(resources)
```
Comment (jsmartt) [inline lib/oneview-sdk/resource.rb]: And while this is a new method, I'd rename it to `ensure_and_get_uris`. This conveys the order of the operations and the plurality of the return values

Comment (ricardogpsf) [inline lib/oneview-sdk/resource.rb]: done. =)

**Pull request merged**